include/maliput/api/lane_boundary.h

```diff
@@ -0,0 +1,188 @@
+///   Boundary N    ...    Boundary 2    Boundary 1    Boundary 0
+///      |                     |             |             |
+///      | Lane N-1  |   ...   |   Lane 1    |   Lane 0    |
+///      |                     |             |             |
+///  (left edge)                                     (right edge)
+/// ```
+///
+/// Where:
+/// - Boundary 0 is the right edge (minimum r coordinate).
+/// - Boundary N is the left edge (maximum r coordinate).
+/// - Boundaries are indexed with increasing r direction.
+///
+/// Each LaneBoundary provides:
+/// - Reference to the lane on its left (if any).
+/// - Reference to the lane on its right (if any).
+/// - Query methods for lane markings at specific s-coordinates.
+///
+/// The design ensures that adjacent lanes share the same boundary object,
+/// avoiding redundancy and ensuring consistency. For example, Lane 1's right
+/// boundary is the same object as Lane 0's left boundary (Boundary 1).
+///
+/// This design is influenced by OpenDRIVE's road marking model and OSI's
+/// lane boundary concept.
+class LaneBoundary {
+ public:
+  MALIPUT_NO_COPY_NO_MOVE_NO_ASSIGN(LaneBoundary)
+
+  /// Unique identifier for a LaneBoundary.
+  using Id = TypeSpecificIdentifier<class LaneBoundary>;
+
+  virtual ~LaneBoundary() = default;
+
+  /// @returns The unique identifier for this LaneBoundary.
+  Id id() const { return do_id(); }
+
+  /// @returns The Segment to which this LaneBoundary belongs.
+  const Segment* segment() const { return do_segment(); }
+
+  /// @returns The index of this boundary within the parent Segment.
+  ///
+  /// Boundaries are indexed from 0 (rightmost, minimum r) to num_lanes()
+  /// (leftmost, maximum r).
+  int index() const { return do_index(); }
+
+  /// @returns The Lane immediately to the left of this boundary (increasing r direction),
+  ///          or nullptr if this is the leftmost boundary of the Segment.
+  const Lane* lane_to_left() const { return do_lane_to_left(); }
+
+  /// @returns The Lane immediately to the right of this boundary (decreasing r direction),
+  ///          or nullptr if this is the rightmost boundary of the Segment.
+  const Lane* lane_to_right() const { return do_lane_to_right(); }
+
+  /// Gets the lane marking at a specific s-coordinate along this boundary.
+  ///
+  /// @param s The s-coordinate along the boundary (in the lane coordinate system).
+  ///          Typically in the range [0, segment_length].
+  ///
+  /// @returns The lane marking at the specified position, or std::nullopt if
+  ///          no marking information is available at that location.
+  ///
+  /// @note The s-coordinate system follows the lane(s) adjacent to this boundary.
+  ///       For consistency, implementations should use the reference lane's
+  ///       s-coordinate (e.g., the lane to the right, or lane to the left if
+  ///       there's no lane to the right).
+  std::optional<LaneMarking> GetMarking(double s) const { return DoGetMarking(s); }
```

Comment:
Probably this should return std::optional<LaneMarkingResult> instead, to also provide information about the s_start and s_end

Reply:
It already does? Or you meant something else?